src/legacy/core_plugins/kibana/public/visualize/np_ready/editor/editor.js

```diff
@@ -116,6 +116,14 @@ function VisualizeAppController(
     dirty: !savedVis.id,
   });
 
+  $scope.isDirty = false;
+
+  vis.on('dirtyStateChange', ({ isDirty }) => {
```

Comment:
Why do we need a separate key on the scope for this? Shouldn't it be enough to `$scope.$digest()` to make angular aware of the change and keep this information within `vis`? Or is this problematic because a digest _could_ be already in progress?

I'm asking because it seems like a step backwards to introduce more angular here.

Reply:
Hmm.. OK, maybe you are right. Updated.
But I didn't see a problem in a previous approach...